src/components/animation/ripple.ts

```diff
@@ -0,0 +1,220 @@
+import { Component, onMounted, useRef, useState } from "@odoo/owl";
+import { Rect, SpreadsheetChildEnv } from "../../types";
+import { css, cssPropertiesToCss, getElementMargins } from "../helpers/css";
+import { getBoundingRectAsPOJO } from "../helpers/dom_helpers";
+
+const RIPPLE_KEY_FRAMES = [
+  { transform: "scale(0)" },
+  { transform: "scale(0.8)", offset: 0.33 },
+  { opacity: "0", transform: "scale(1)", offset: 1 },
+];
+
+css/* scss */ `
+  .o-ripple {
+    z-index: 1;
+  }
+`;
+
+interface RippleProps {
+  color: string;
+  opacity: number;
+  duration: number;
+
+  /** If true, the ripple will play from the element center instead of the position of the click */
+  ignoreClickPosition?: boolean;
+
+  /** Width of the ripple. Defaults to the width of the element the ripple is on (without margins). */
+  width?: number;
+  /** Height of the ripple. Defaults to the height of the element the ripple is on (without margins). */
+  height?: number;
+
+  offsetY?: number;
+  offsetX?: number;
+
+  allowOverflow?: boolean;
+  enabled: boolean;
+  onAnimationEnd: () => void;
+  class: string;
+}
+
+interface RippleEffectProps
+  extends Omit<Required<RippleProps>, "ignoreClickPosition" | "enabled" | "class"> {
+  x: string;
+  y: string;
+}
+
+interface RippleDef {
+  rippleRect: Rect | undefined;
+  id: number;
+}
+
+interface RippleState {
+  ripples: Array<RippleDef>;
+}
+
+interface RectWithMargins extends Rect {
+  marginTop: number;
+  marginLeft: number;
+}
+
+class RippleEffect extends Component<RippleEffectProps, SpreadsheetChildEnv> {
+  static template = "o-spreadsheet-RippleEffect";
+  private rippleRef = useRef("ripple");
+
+  setup() {
+    onMounted(() => {
+      const rippleEl = this.rippleRef.el;
+      if (!rippleEl || !rippleEl.animate) return;
+      const animation = rippleEl.animate(RIPPLE_KEY_FRAMES, {
+        duration: this.props.duration,
+        easing: "ease-out",
+      });
+      animation.onfinish = () => this.props.onAnimationEnd();
```

Comment:
there isn't an API to add an event listener callback such as "addEventLIsntener"?
It would be safer to remove the event listener when the component is unmounted

Reply:
There is an `onAnimationEnd` that I tried to use, but it doesn't seems to work with animations fired from JS :/

Reply:
Nvm you're right there is a `addEventListener` in animiations